content/courses/Ada_For_The_C_Embedded_Developer/chapters/02_Perspective.rst

```diff
@@ -2759,3 +2776,96 @@ operator as a function, enclose its "name" in quotes:
        end if;
     end Main;
 
+.. _Aspects:
+
+Aspects
+~~~~~~~
+
+Aspect specifications allow you to define certain characteristics of a
+declaration using the :ada:`with` keyword after the declaration:
+
+.. code-block:: ada
+
+    procedure Some_Procedure is <procedure_definition>
+      with Some_Aspect => <aspect_specification>;
+
+    function Some_Function is <function_definition>
+      with Some_Aspect => <aspect_specification>;
+
+    type Some_Type is <type_definition>
+      with Some_Aspect => <aspect_specification>;
+
+    Obj : Some_Type with Some_Aspect => <aspect_specification>;
+
+For example, you can inline a subprogram by specifying the :ada:`Inline`
+aspect:
+
+.. code:: ada no_button project=Courses.Ada_For_C_Embedded_Dev.Perspective.Inline_Aspect
+
+    package Float_Arrays is
+
+       type Float_Array is array (Positive range <>) of Float;
+
+       function Average (Data : Float_Array) return Float
+         with Inline;
+
+    end Float_Arrays;
+
+We'll discuss inlining :ref:`later in this course <Inlining>`.
+
+Aspect specifications were introduced in Ada 2012. In previous versions of Ada,
+you had to use a :ada:`pragma` instead. The previous example would be written
+as follows:
+
+.. code:: ada no_button project=Courses.Ada_For_C_Embedded_Dev.Perspective.Inline_Aspect
+
+    package Float_Arrays is
+
+       type Float_Array is array (Positive range <>) of Float;
+
+       function Average (Data : Float_Array) return Float;
+
+       pragma Inline (Average);
+
+    end Float_Arrays;
+
+Aspects and attributes might refer to the same kind of information. For
+example, we can use the :ada:`Size` aspect to define the expected size of
```

Comment:
Yes, since we're explaining this topic later in another chapter, I'd prefer to keep it simple in this section and just add this word ("minimum") as you're suggesting.

Reply:
By the way, I've just re-read the section where we explain the `Size` aspect (admonition "Details about `'Size`" in chapter 4), and it seems to me that we're explaining the topic very well there. Let me know if you think anything is missing that section of chapter 4.